### DISCUSSION TITLE
Is there a way to generate an api for a pre-existing custom resource.

### DISCUSSION THREAD

**Discussion by empath-nirvana** [kube-rs/kube#636] [Q&A] [answered]
Sorry if this is a stupid question, but if I've got a pre-existing custom resource installed on the cluster, like for example an ArgoCD application -- is there a way to generate an api for that without writing out the whole struct myself if I wanted to create or modify those objects?

**Comment by clux** [comment, marked as answer]
Yeah, that is possible. You can use one of the Api constructors with the _with suffix for dynamic types. You would have to supply it with dynamic type information typically through ApiResource (either manually or through the discovery module).

There are some relevant examples on the examples folder starting with dynamic_ that would both be relevant here :-)

**Comment by nightkr** [comment]
To be clear, this will let you read or write those objects, but we currently don't have a way to generate a *type-safe* API for CRDs.
